CriticalMass/UIView+Nib.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  UIView+Nib.swift
+//  CriticalMaps
+//
+//  Created by Malte Bünz on 28.03.19.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import UIKit
+
+protocol NibInstantiatable {
+    static func nibName() -> String
+}
+
+extension NibInstantiatable {
+    static func nibName() -> String {
+        return String(describing: self)
+    }
+}
+
+extension NibInstantiatable where Self: UIView {
+    
+    @discardableResult
+    static func fromNib() -> Self? {
+        let bundle = Bundle(for: self)
+        let nib = bundle.loadNibNamed(nibName(), owner: self, options: nil)
+        guard let view = nib?.first as? Self else {
```

Comment:
Minor: we don’t need a guard here because the expected return type is an optional

Reply:
`nib?.first` is of type `Any?` and a conversion to `Self` is not possible. So I have to use conditional cast

Reply:
```swift
return nib?.first as? Self
```
compiles for me

Reply:
It does here aswell 😀
Just thought it is more explicit. Will change to shorter version